### PR TITLE
[v4] [core] fix(Switch): fix text color and indicator border

### PR DESCRIPTION
**Pull request by johnly13**
#### Checklist

- [ ] Includes tests
- [ ] Update documentation

#### Changes proposed in this pull request:

* Uses normal button `box-shadow` instead of `inset` `box-shadow` for dark switch indicators
* Fixes text colors for switches
* Removes borders for disabled switch indicator
* Fixes dark switch indicator background colors

#### Screenshot

| Before | After |
| --- | --- |
| Screen Shot 2021-12-03 at 15 32 47 | Screen Shot 2021-12-06 at 14 32 02|
| Screen Shot 2021-12-03 at 15 32 52 | Screen Shot 2021-12-06 at 14 32 08 |
| Screen Shot 2021-12-03 at 15 32 57 | Screen Shot 2021-12-06 at 17 17 44|
| Screen Shot 2021-12-03 at 15 33 03 | Screen Shot 2021-12-06 at 17 17 49 |

(note: I think the indicator looks off center vertically in some of these screenshots. I think this is an artifact of zooming in + screenshotting, rather than how they actually are styled)

### CODE REVIEW

File: packages/core/src/components/button/_common.scss
Comment (johnly13) [inline packages/core/src/components/button/_common.scss:51]: using variables instead of hardcoded values

File: packages/core/src/components/forms/_controls.scss
Comment (johnly13) [inline packages/core/src/components/forms/_controls.scss:303]: Explicitly setting text colors for switch's inner text (for light/dark and enabled/disabled)

File: packages/core/src/components/forms/_controls.scss
Comment (johnly13) [inline packages/core/src/components/forms/_controls.scss]: Explicitly set dark switch indicator's box shadow to not be inset, to be consistent with other indicator box shadows.

File: packages/core/src/components/forms/_controls.scss
Comment (johnly13) [inline packages/core/src/components/forms/_controls.scss:431]: since the box shadow is different now, we don't get this blurring effect anymore without the inset

File: packages/core/src/blueprint.scss
Comment (johnly13) [inline packages/core/src/blueprint.scss:278]: Removed these overrides, now built in the backgrounds into switch variables

File: packages/core/src/components/forms/_controls.scss
Comment (johnly13) [inline packages/core/src/components/forms/_controls.scss:368]: no borders for disabled switch indicator

**Comment by johnly13** [comment]
@adidahiya this PR should be ready for review now!

**Review by adidahiya — APPROVED** [review]
this is kind of an an explosion of Sass variables... probably unnecessary complexity which makes the code harder to work with... but since that's kind of the existing code pattern for these styles, we can leave it as-is and try to refactor in a separate PR 🙃 

code changes look fine, and visuals lgtm... I'll wait for @aycai for final design approval

**Comment by johnly13** [comment]
yeah, I was pretty sad adding all these new variables, but given that the switch styles have diverged slightly from the button styles, it didn't make sense to use the existing button variables. I'll think of ways to improve this in a follow up - I think low hanging fruit is to remove variables that are only used once (which I'd imagine there are a few of)

> this is kind of an an explosion of Sass variables... probably unnecessary complexity which makes the code harder to work with... but since that's kind of the existing code pattern for these styles, we can leave it as-is and try to refactor in a separate PR 🙃
> 
> code changes look fine, and visuals lgtm... I'll wait for @aycai for final design approval

**Comment by aycai** [comment]
@johnly13 Shoot, didn't show switch with text in the mocks - light mode + enabled + disabled state text should be 60% opacity to match other states, but rest looks good!

**Comment by johnly13** [comment]
> @johnly13 Shoot, didn't show switch with text in the mocks - light mode + enabled + disabled state text should be 60% opacity to match other states, but rest looks good!

Done - updated screenshots above as well!

**Pull request merged**
